src/core/main.c

```diff
@@ -2603,6 +2603,11 @@ int main(int argc, char *argv[]) {
                 /* clear the kernel timestamp,
                  * because we are not PID 1 */
                 kernel_timestamp = DUAL_TIMESTAMP_NULL;
+
+                if (detect_container() <= 0 && mac_selinux_init() < 0) {
```

Comment:
Hmm, why the "detect_container()" check here? isn't that redundant? The other calls in our codebase generally don't do that either()?

Reply:
I used it because the pid1 instance also calls `mac_selinux_init()` https://github.com/systemd/systemd/blob/0389f4fa81e93be298721ce699889161fbe4046b/src/core/main.c#L2561 only if `detect_container() <= 0` https://github.com/systemd/systemd/blob/0389f4fa81e93be298721ce699889161fbe4046b/src/core/main.c#L2529

Reply:
well, we do the detect_container() check there because of the other calls we do in its `{}` block, not because of the selinux init call...

I don't think we need to check for a container here, let's simplify this, mac_selinux_init() should work in containers too, because it early on calls is_selinux_enabled() which internally does the selinuxfs ro check, which detects containers sufficiently.

Reply:
changed